plugins/serverless/src/tasks/teardown.ts

```diff
@@ -0,0 +1,35 @@
+import { ToolKitError } from '@dotcom-tool-kit/error'
+import { hookFork, styles, waitOnExit } from '@dotcom-tool-kit/logger'
+import { Task } from '@dotcom-tool-kit/types'
+import { ServerlessSchema } from '@dotcom-tool-kit/types/lib/schema/serverless'
+import { readState } from '@dotcom-tool-kit/state'
+import { spawn } from 'child_process'
+
+export default class ServerlessTeardown extends Task<typeof ServerlessSchema> {
+  static description = 'Teardown existing serverless functions'
+
+  async run(): Promise<void> {
+    const { configPath, systemCode, regions } = this.options
+
+    const reviewState = readState('review')
```

Comment:
**question**: why do we need to store the stage name in state? we have access to the same environment variables so can calculate the stage name the same way the `ServerlessProvision` step does. we could define a shared function in the serverless plugin to make sure they always use the same logic to select a name!

Reply:
ah, of course, good point! what format is `CIRCLECI_WORKFLOW_ID` in? could we compress it in some way (e.g., base64 encoding) to get it to fit? not sure if one approach (compressed workflow id vs written build number) is preferable over the other...